R/preprocessing_pipeline.R

```diff
@@ -0,0 +1,53 @@
+## pipeline to preprocess the corpus
+## TODO modular set your own pipeline
+stopwordprocess <- function(removeStopwords, custom_stopwords = NULL, language){
+    if(removeStopwords){
+        if(!is.null(custom_stopwords)){
+            if(is.character(custom_stopwords)){
+                removeStopwords <- custom_stopwords
+            }else{
+                stop("ERROR: custom_stopwords must be formatted as a character vector of strings. Such as stopwords::stopwords()")
+            }
+        }else{
+            removeStopwords <-  quanteda::stopwords(language = language)}}
+    return(removeStopwords)
+}
+
+preprocessing_pipeline <- function(choices  ,
+                                   text,
+                                   infrequent_term_threshold = .01,
+                                   verbose = FALSE,
+                                   save_dfm = FALSE,
+                                   intermediate_directory = NULL,
+                                   custom_stopwords = NULL){
+    
+    quanteda_options(language_stemmer = meta(text)$language)
+    stopwords <- stopwordprocess(choices$removeStopwords, custom_stopwords, meta(text)$language)
+    if(choices$use_ngrams){ #need to tokenize before ngram before preprocessing
+        text <- quanteda::tokens(text)
+        text <- quanteda::tokens_ngrams(text, n = 1:3)
+    }
+
+    text <- quanteda::dfm(text,
+                       tolower = choices$lowercase,
+                       stem = choices$stem,
```

Comment:
If this were done separately as `tokens_wordstem()` then you could add language support for the stemmer too.

Reply:
the stemmer honors `quanteda_options(language_stemmer = meta(text)$language)` but that would be more elegant.